src/contracts/exchanges/third-party/kyber/transactions/setBaseRate.ts

```diff
@@ -44,25 +51,29 @@ const splitArray = (arr: number[], length: number): number[][] => {
 
 type SetBaseRateResult = boolean;
 
+// If buy-sell prices are not explicity passed, assumes passed price to be sell
 const prepareArgs: PrepareArgsFunction<SetBaseRateArgs> = async (
   environment,
   { prices, blockNumber: givenBlockNumber },
 ) => {
   environment.logger('debug', LogLevels.DEBUG, prices);
 
   const numberOfTokens = prices.length;
+  let enhancedPrices: BuySell[] = [];
```

Comment:
You could rewrite this as:

```ts
const enhancedPrices: BuySell[] = isBuySell(prices) 
  ? prices 
  : prices.map(p => ({
      buy: createPrice(p.quote, p.base),
      sell: p,
    }))
```

To avoid the `let` statement.

But it's ok how it is.

Reply:
Fixed